src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import core.basesyntax.file.DataConverter;
+import core.basesyntax.file.DataConverterImpl;
+import core.basesyntax.file.FileReader;
+import core.basesyntax.file.FileReaderImpl;
+import core.basesyntax.file.FileWriter;
+import core.basesyntax.file.FileWriterImpl;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ReportGeneratorImpl;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.ShopServiceImpl;
+import core.basesyntax.strategy.BalanceOperation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.PurchaseOperation;
+import core.basesyntax.strategy.ReturnOperation;
+import core.basesyntax.strategy.SupplyOperation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        // 1. Read the data from the input CSV file
+        FileReader fileReader = new FileReaderImpl();
+        List<String> inputReport = fileReader.read(
+                "src/main/java/core/basesyntax/reportToRead.csv");
+
+        // 2. Convert the incoming data into FruitTransactions list
+
+        // 3. Create and feel the map with all OperationHandler implementations
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+        ShopService shopService = new ShopServiceImpl(operationStrategy);
+        operationHandlers.put(FruitTransaction.Operation.BALANCE,
+                new BalanceOperation(shopService));
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperation(shopService));
+        operationHandlers.put(FruitTransaction.Operation.RETURN,
+                new ReturnOperation(shopService));
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyOperation(shopService));
+
+        // 4. Process the incoming transactions with applicable OperationHandler implementations
+        DataConverter dataConverter = new DataConverterImpl();
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+        shopService.process(transactions);
+
+        // 5.Generate report based on the current Storage state
+        ReportGenerator reportGenerator = new ReportGeneratorImpl();
+        String resultingReport = reportGenerator.getReport(shopService);
+
+        // 6. Write the received report into the destination file
+        FileWriter fileWriter = new FileWriterImpl();
+        fileWriter.write(resultingReport, "src/main/java/core/basesyntax/finalReport.csv");
```

Comment:
Verify that the file path 'src/main/java/core/basesyntax/finalReport.csv' is correct and that you have write permissions to this location. Otherwise, the `FileWriter` may throw an error.